vsphere/datadog_checks/vsphere/cache.py

```diff
@@ -14,12 +18,14 @@ class VSphereCache(object):
     """
 
     def __init__(self, interval_sec):
-        self._last_ts = 0
+        # type: (int) -> None
+        self._last_ts = 0.0
         self._interval = interval_sec
-        self._content = {}
+        self._content = {}  # type: Dict[Any, Any]
 
     @contextmanager
     def update(self):
+        # type: () -> Generator[None, None, None]
```

Comment:
> Be liberal in what you accept, and conservative in what you send.
> -- Postel's Law

It's usually preferable to return the most precise type possible and let the client decide what type/subtype to use.

Reply:
And in this case, using a Generator seems a bit more correct semantically since we are using `update` as context manager, not something we want to "iterate" over.